src/index.js

```diff
@@ -0,0 +1,127 @@
+'use babel';
+
+// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
+import { CompositeDisposable } from 'atom';
+import path from 'path';
+import escapeHtml from 'escape-html';
+import pluralize from 'pluralize';
+import * as helpers from 'atom-linter';
+
+const DEFAULT_ARGS = [
+  '--cache', 'false',
+  '--force-exclusion',
+  '--format', 'json',
+  '--display-style-guide',
+];
+
+const formatMessage = ({ message, cop_name: copName, url }) => {
+  const formattedMessage = escapeHtml(message || 'Unknown Error');
+  let formattedCopName = '';
+  if (copName && url) {
+    formattedCopName = ` (<a href="${escapeHtml(url)}">${escapeHtml(copName)}</a>)`;
+  } else if (copName) {
+    formattedCopName = ` (${escapeHtml(copName)})`;
+  }
+  return formattedMessage + formattedCopName;
+};
+
+const parseFromStd = (stdout, stderr) => {
+  let parsed;
+  try {
+    parsed = JSON.parse(stdout);
+  } catch (error) {
+    // continue regardless of error
+  }
+  if (typeof parsed !== 'object') { throw new Error(stderr || stdout); }
+  return parsed;
+};
+
+const forwardRubocopToLinter =
+  ({ message: rawMessage, location, severity, cop_name }, filePath) => {
+    const [message, url] = rawMessage.split(/ \((.*)\)/, 2);
+    const { line, column, length } = location || { line: 1, column: 1, length: 0 };
```

Comment:
Again, if `location` isn't specified you would be better not specifying a `range` attribute at all on the message since this default will result in a zero length range which is invisible in the editor (but still "there"!).